### PR TITLE
refactor of missing video encodings job

### PR DESCRIPTION
**Pull request by NathanielBennett**
This is a refactor of this: https://github.com/guardian/frontend/pull/8154 which does three things: 

1) Uses a slightly different api query: http://content.guardianapis.com/search?api-key=test&show-assets=all&tag=type/video&page-size=100 and uses the assets parse sub-document to locate the encodings for each video encoding. I've updated some variable names and tidied up the indentation because the scala for this is a little bit perverted ... 

2) Strips out any newline characters in the individual encoding urls: Some of the octopus urls have newline characters embedded in them which (a) causes the agent to throw error and (b) is a possible cause for video assents being unavailable in the first place

3) Hardcodes the name of of the SNS topic which missing encoding notifications get posted to. This is because previously, the topic created by cloudformation gets recreated each time someone cloudforms.

### CODE REVIEW

**Comment by tinmaungnaing** [comment]
Thankyou all friend

File: admin/app/jobs/MissingVideoEncodingsJob.scala [hunk 1]
```diff
@@ -41,44 +37,45 @@ object VideoEncodingsJob extends ExecutionContexts with Logging  {
...
-         response.results map {
-           Video(_)
-         }
+        response =>
+            response.results map { Video(_)  }
      }
 
-     val videos = Await.result(apiVideoResponse.map{ actualVideo =>
-       actualVideo map ( video => video )
+     val allVideoContent = Await.result(apiVideoResponse.map{ actualVideo =>
```
Comment (gklopper) [inline admin/app/jobs/MissingVideoEncodingsJob.scala]: Does this really need to block?

**Comment by NathanielBennett** [comment]
@gklopper No there wasn't a good reason for blocking - so I've taken that out. I now have nested futures, though.

**Comment by gklopper** [comment]
:+1:

**Pull request merged**